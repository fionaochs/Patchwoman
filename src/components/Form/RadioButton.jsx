import React from 'react';
import PropTypes from 'prop-types';

export const RadioButton = ({ name, value, onChange }) => {
  return (
    <>
      <input
        id={value}
        type="radio"
        name={name}
        value={value}
        onChange={onChange} />
      <label htmlFor={value}>{value}</label>
    </>
  );
};
RadioButton.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  name: PropTypes.string
};
