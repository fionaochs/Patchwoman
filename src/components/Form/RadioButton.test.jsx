  
import React from 'react';
import { shallow } from 'enzyme';
import { RadioButton } from './RadioButton.jsx';

describe('RadioButton', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <RadioButton
        id="get"
        onChange={() => {}}
        name="method"
        method="get"
        value="get" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
