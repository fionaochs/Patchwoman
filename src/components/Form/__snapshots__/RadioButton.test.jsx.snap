// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton matches a snapshot 1`] = `
<Fragment>
  <input
    id="get"
    name="method"
    onChange={[Function]}
    type="radio"
    value="get"
  />
  <label
    htmlFor="get"
  >
    get
  </label>
</Fragment>
`;
